syntax = "proto3";

package orchestrator_service;

import "imu_service.proto";

service OrchestratorService {
  // HealthCheck is used to check the health of the orchestrator service.
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
  
  // OrchestratorStatus is used to send orchestrator status updates.
  rpc OrchestratorStatus(OrchestratorStatusRequest) returns (OrchestratorStatusResponse);

  // Receive data from sensors.
  rpc ReceiveIMUData(imu_service.IMUPayload) returns (imu_service.IMUPayloadResponse);
}

message HealthCheckRequest {}

message HealthCheckResponse {
  bool status = 1;
}

message OrchestratorStatusRequest {}

message OrchestratorStatusResponse {
  bool is_ready = 1; // Indicates if the orchestrator is ready to process requests
  string current_activity = 2; // Name of the current activity being processed
}
